### PR TITLE
More cleanup to LinkCrawler.kt by using more Kotlin features, etc.

### PR DESCRIPTION
**Pull request by afollestad**
Also improved some formatting.

### CODE REVIEW

File: library/src/main/java/net/vrgsoft/library/LinkCrawler.kt [hunk 1]
```diff
@@ -79,42 +83,49 @@ class LinkCrawler {
...
-                            content.title == "" -> {
-                                val matchTitle = Regex.match(content.htmlCode, Regex.TITLE_PATTERN, 2)
-                                if (matchTitle != "") {
-                                    content.title = htmlDecode(matchTitle)
-                                }
-                            }
+                        content.title = metaTags["title"] ?: ""
+                        content.description = metaTags["description"] ?: ""
+
+                        if(content.title.isEmpty()) {
```
Comment (msfjarvis) [inline library/src/main/java/net/vrgsoft/library/LinkCrawler.kt]:
```suggestion
                        if (content.title.isEmpty()) {
```

File: library/src/main/java/net/vrgsoft/library/LinkCrawler.kt [hunk 2]
```diff
@@ -79,42 +83,49 @@ class LinkCrawler {
...
+                        if (content.description.isEmpty()) {
                             content.description = crawlCode(content.htmlCode)
                         }
                         content.description = content.description.replace(Regex.SCRIPT_PATTERN, "")
-                        when {
-                            metaTags["image"] != "" -> content.images.add(
-                                    metaTags["image"]!!)
-                            else -> content.images = getImages(doc).toMutableList()
+
+                        if(metaTags["image"]?.isNotEmpty() == true) {
```
Comment (msfjarvis) [inline library/src/main/java/net/vrgsoft/library/LinkCrawler.kt:111]:
```suggestion
                        if (metaTags["image"]?.isNotEmpty() == true) {
```

**Review by msfjarvis — APPROVED** [review]

File: library/src/main/java/net/vrgsoft/library/LinkCrawler.kt (hunk 2 above)
Comment (afollestad) [reply library/src/main/java/net/vrgsoft/library/LinkCrawler.kt:111]: @MSF-Jarvis that's not valid syntax, the `?` makes it return `Boolean?` so an equality check is necessary

Comment (msfjarvis) [reply library/src/main/java/net/vrgsoft/library/LinkCrawler.kt:111]: > @MSF-Jarvis that's not valid syntax, the `?` makes it return `Boolean?` so an equality check is necessary

Oh true, didn't notice the nullable type. Will cut this down to just the formatting.

**Pull request closed without merging**